src/control/server/iosrv.go

```diff
@@ -359,8 +381,16 @@ func (srv *iosrv) startCmd() error {
 	}
 
 	srv.cmd = exec.Command(binPath, srv.config.Servers[srv.index].CliOpts...)
-	srv.cmd.Stdout = os.Stdout
-	srv.cmd.Stderr = os.Stderr
+	srv.cmd.Stdout = &cmdLogger{
+		logFn:  srv.log.Info,
+		prefix: fmt.Sprintf("%s:%d", ioServerBin, srv.index),
+	}
+	srv.cmd.Stderr = &cmdLogger{
+		logFn:  srv.log.Error,
+		prefix: fmt.Sprintf("%s:%d", ioServerBin, srv.index),
+	}
+	// FIXME(DAOS-3105): This shouldn't be the default. The command environment
```

Comment:
I thought this in the past and still agree but unfortunately it's not trivial due to some of the dependencies of the IO server. needs investigation.

Reply:
Agreed, which is why I made it a FIXME with a ticket so that it doesn't get lost. I think that longer term we should not support passing configuration through external environment because it introduces a great deal of uncertainty about how things were actually configured, making support more difficult.